src/graph/InsertVertexExecutor.cpp

```diff
@@ -124,55 +142,101 @@ StatusOr<std::vector<storage::cpp2::Vertex>> InsertVertexExecutor::prepareVertic
-            auto iter = schema->begin();
-            while (iter) {
-                // Check value type
-                auto schemaType = iter->getType();
-                uint32_t fieldIndex = schemaIndexes_[index][iter->getName()] + valuePos;
-                if (fieldIndex >= values.size()) {
-                    return Status::Error("Wrong index of `%s'", iter->getName());
-                }
-                auto &value = values[fieldIndex];
-                // Check value type
-                if (!checkValueType(schemaType, value)) {
-                    auto *output = "ValueType is wrong, schema type [%d], "
-                                   "input type [%d], near `%s'";
-                    auto error = folly::stringPrintf(output, static_cast<int32_t>(schemaType.type),
-                            value.which(), iter->getName());
-                    LOG(ERROR) << error;
-                    return Status::Error(std::move(error));
-                }
-                if (schemaType.type == nebula::cpp2::SupportedType::TIMESTAMP) {
-                    auto timestamp = toTimestamp(value);
-                    if (!timestamp.ok()) {
-                        return timestamp.status();
+            auto schemaNumFields = schema->getNumFields();
+            for (size_t schemaIndex = 0; schemaIndex < schemaNumFields; schemaIndex++) {
+                auto fieldName = schema->getFieldName(schemaIndex);
+                auto positionIter = propsPosition.find(fieldName);
+
+                auto schemaType = schema->getFieldType(schemaIndex);
+                if (positionIter != propsPosition.end()) {
+                    auto position = propsPosition[fieldName];
+                    auto value = values[position + valuePosition];
+                    if (!checkValueType(schemaType, value)) {
+                       LOG(ERROR) << "ValueType is wrong, schema type "
+                                   << static_cast<int32_t>(schemaType.type)
+                                   << ", input type " <<  value.which();
+                        return Status::Error("ValueType is wrong");
+                    }
+
+                    if (schemaType.type == nebula::cpp2::SupportedType::TIMESTAMP) {
+                       auto timestamp = toTimestamp(value);
+                        if (!timestamp.ok()) {
+                            return timestamp.status();
+                        }
+                        writeVariantType(writer, timestamp.value());
+                    } else {
+                        writeVariantType(writer, value);
                     }
-                    writeVariantType(writer, timestamp.value());
+
                 } else {
-                    writeVariantType(writer, value);
-                }
+                    // fetch default value from cache
+                    VariantType value;
+                    auto type = schema->getFieldType(schemaIndex).type;
+                    switch (type) {
```

Comment:
It seems this piece is same with InsertEdge, could we reuse them?

Reply:
Thanks & I will move it into `Executor`